File: backend/routers/elections.py
from fastapi import APIRouter, HTTPException
from services import election_data

router = APIRouter()

@router.get("/")
def get_all_elections():
    """Return all elections for the Explore page."""
    return election_data.get_all_elections()

@router.get("/phases/{country_code}")
def get_phases(country_code: str):
    data = election_data.get_election_phases(country_code)
    if "error" in data:
        raise HTTPException(status_code=404, detail=data["error"])
    return data

@router.get("/supported-countries")
def get_supported_countries():
    return [
        {"code": "us", "name": "United States"},
        {"code": "uk", "name": "United Kingdom"},
        {"code": "in", "name": "India"}
    ]

File: Elec-main/src/backend/routes/elections.ts
import { Router, Request, Response } from "express";
import { getElectionPhases } from "../services/electionData";

const router = Router();

router.get("/", (req: Request, res: Response) => {
  res.json({ countries: ["us", "uk", "india", "canada", "australia"] });
});

router.get("/:country", (req: Request, res: Response) => {
  const data = getElectionPhases(req.params.country);
  if (!data) {
    return res.status(404).json({ detail: `No election data for: ${req.params.country}` });
  }
  res.json(data);
});

export default router;
